src/components/AvatarCropModal/AvatarCropModal.js

```diff
@@ -0,0 +1,258 @@
+import variables from '../../styles/variables';
+import Button from '../Button';
+import HeaderWithCloseButton from '../HeaderWithCloseButton';
+import Icon from '../Icon';
+import * as Expensicons from '../Icon/Expensicons';
+import Modal from '../Modal';
+import Text from '../Text';
+import withLocalize, {withLocalizePropTypes} from '../withLocalize';
+import withWindowDimensions, {windowDimensionsPropTypes} from '../withWindowDimensions';
+import ImageCropView from './components/ImageCropView';
+import Slider from './components/Slider';
+import imageManipulator from './libs/imageManipulator';
+
+const propTypes = {
+    /** Link to image for cropping   */
+    imageUri: PropTypes.string,
+
+    /** Callback to be called when user closes the modal */
+    onClose: PropTypes.func,
+
+    /** Callback to be called when user crops image */
+    onCrop: PropTypes.func,
+
+    ...withLocalizePropTypes,
+
+    ...windowDimensionsPropTypes,
+};
+
+const defaultProps = {
+    imageUri: null,
+    onClose: () => {},
+    onCrop: () => {},
+};
+
+// This component cant be writen using class since reanimated API uses hooks
+const AvatarCropModal = memo((props) => {
+    const imageWidth = useSharedValue(1);
+    const imageHeight = useSharedValue(1);
+    const translateY = useSharedValue(0);
+    const translateX = useSharedValue(0);
+    const scale = useSharedValue(1);
+    const rotation = useSharedValue(0);
+    const translateSlider = useSharedValue(0);
+
+    const containerSize = props.isSmallScreenWidth ? Math.min(props.windowWidth, 500) - 40 : variables.sideBarWidth - 40;
+    const sliderLineWidth = containerSize - 105;
+
+    const initializeImage = useCallback(() => {
+        translateY.value = 0;
+        translateX.value = 0;
+        scale.value = 1;
+        rotation.value = 0;
+        translateSlider.value = 0;
+    }, []);
+
+    useEffect(() => {
+        if (!props.imageUri) { return; }
+        initializeImage();
+        Image.getSize(props.imageUri, (width, height) => {
+            imageHeight.value = height;
+            imageWidth.value = width;
+            translateSlider.value += 0.01;
+            rotation.value += 360;
+        });
+    }, [props.imageUri, initializeImage]);
+
+    const panGestureEvent = useAnimatedGestureHandler({
+        onStart: (_, context) => {
+            // eslint-disable-next-line no-param-reassign
+            context.translateX = translateX.value;
+            // eslint-disable-next-line no-param-reassign
+            context.translateY = translateY.value;
+        },
+        onActive: (event, context) => {
+            let heighRatio = 1.0;
+            let widthRation = 1.0;
+            if (imageWidth.value > imageHeight.value) {
+                heighRatio = imageWidth.value / imageHeight.value;
+            } else {
+                widthRation = imageHeight.value / imageWidth.value;
+            }
+
+            const radius = containerSize / 2;
```

Comment:
Please add a comment for why this value is what it is.

Reply:
Renamed radius to apothem and added a comment.